internal/dockerfile2dot/convert.go

```diff
@@ -46,6 +56,9 @@ func dockerfileToSimplifiedDockerfile(content []byte) SimplifiedDockerfile {
 
 			simplifiedDockerfile.Stages = append(simplifiedDockerfile.Stages, stage)
 
+			// set layer index as stage index
+			layerIndex=0
+
```

Comment:
Hi Patrick, why do you think the layers `FROM` should be added? those layers are being added as stages. If we add those layers, then in the main graph we will have a node for the stage (standard graph) and the same node with the command FROM as a layer.

Reply:
Yes, I definitely think we should go for what you proposed in #82 🙂 

Having said that, I'm not sure if we can fit all of it into this PR. After looking into my implementation — it's been a few months since I wrote it 😄 — I think I'll have to refactor some of the core logic. The main issue right now is that the `waitFor` [dependencies](https://github.com/patrickhoefler/dockerfilegraph/blob/main/internal/dockerfile2dot/structs.go#L15) — the edges in the graph — are currently between stages, but for the new feature they need to be between layers.

My suggestion therefore would be:

1. We see how far we can take this PR based on what you have already implemented without any major additional refactoring.
2. We merge the PR.
3. I refactor the core logic.
4. We make another PR to bring it in line with your suggestion in #82.

What do you think?

Reply:
I have replicated the example using the duplicated node as a reference for the edge, but with a style=invis. I know is not the most perfect solution, but I think might be good as a first proposal.

![Example output including layer per stage](https://user-images.githubusercontent.com/35694200/168588978-d0386eaf-e79b-45d8-a6e1-b99a4771ca05.png)

Reply:
> The `FROM` layers should be added just like all other layers.

Done in #https://github.com/patrickhoefler/dockerfilegraph/pull/88/commits/d9bbf388138436d1878e0b0f3fa58ec236076263

Reply:
Thanks Daniel, looks great!